File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/gui/screen/ScreenManager.java
package pize.tests.minecraft.game.gui.screen;

import pize.app.Disposable;
import pize.app.Resizable;
import pize.graphics.util.batch.TextureBatch;

import java.util.HashMap;
import java.util.Map;

public class ScreenManager implements Resizable, Disposable{

    private final Map<String, Screen> screens;
    private String current;

    public ScreenManager(){
        screens = new HashMap<>();
    }

    public void setCurrentScreen(String id){
        if(screens.containsKey(id)){
            current = id;
            screens.get(id).onShow();
        }else
            current = null;
    }


    public void putScreen(String id, Screen screen){
        screens.put(id, screen);
    }

    public void putGui(String id, Screen screen){
        screens.put(id, screen);
    }

    public void render(TextureBatch batch){
        Screen screen = screens.get(current);
        if(screen != null){
            screen.update(batch);
            screen.render(batch);
        }
    }

    @Override
    public void resize(int width, int height){
        for(Screen screen: screens.values())
            screen.resize(width, height);
    }

    @Override
    public void dispose(){
        for(Screen screen: screens.values())
            screen.dispose();
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/gui/screen/ScreenManager.kt
package pize.tests.minecraft.game.gui.screen

import pize.app.Disposable
import pize.app.Resizable
import pize.graphics.util.batch.TextureBatch

class ScreenManager : Resizable, Disposable {
    private val screens: MutableMap<String?, Screen>
    private var current: String? = null

    init {
        screens = HashMap()
    }

    fun setCurrentScreen(id: String?) {
        if (screens.containsKey(id)) {
            current = id
            screens[id]!!.onShow()
        } else current = null
    }

    fun putScreen(id: String?, screen: Screen) {
        screens[id] = screen
    }

    fun putGui(id: String?, screen: Screen) {
        screens[id] = screen
    }

    fun render(batch: TextureBatch) {
        val screen = screens[current]
        if (screen != null) {
            screen.update(batch)
            screen.render(batch)
        }
    }

    override fun resize(width: Int, height: Int) {
        for (screen in screens.values) screen.resize(width, height)
    }

    override fun dispose() {
        for (screen in screens.values) screen.dispose()
    }
}
